framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java

```diff
@@ -257,6 +257,7 @@ private void updateAtmWithLub(AnnotatedTypeMirror sourceCodeATM, AnnotatedTypeMi
 
         switch (sourceCodeATM.getKind()) {
             case TYPEVAR:
+                // TODO: It seems wrong to LUB the type parameter bounds.
```

Comment:
Can you expand on why it seems wrong?  Also, I think these are uses of type variables rather than type parameter declarations.  So they aren't invariant as you suggest in the comment below.